### PR TITLE
Update JAX-export tutorial

### PR DESCRIPTION
**Pull request by pavithraes**
This PR updates "Exporting StableHLO from JAX" tutorial. Specifically:

- Move from `jax.experimental.export` to `jax.export`, mainly involves explicit use of `jax.jit`
- Modification of narrative text for accuracy, consistency, and additional links
- Add a couple of introductory sentences

### CODE REVIEW

File: docs/tutorials/jax-export.ipynb [hunk 1]
```diff
@@ -1,445 +1,479 @@
...
-          "name": "stdout",
-          "text": [
-            "Result shape: (1, 512, 7, 7)\n"
-          ]
-        }
-      ]
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/Users/pavithraes/mambaforge/envs/shlo-docs/lib/python3.11/pty.py:89: RuntimeWarning: os.fork() was called. os.fork() is incompatible with multithreaded code, and JAX is multithreaded, so this will likely lead to a deadlock.\n",
```
Comment (GleasonK) [inline docs/tutorials/jax-export.ipynb:382]: image

Can we see if there's a way to suppress errors? Not sure if there's a way to write this code without some implicit call to fork, but don't want to show your user path here 🙂

**Review by GleasonK — CHANGES_REQUESTED** [review]
Looks great, thank you! One thing to fix then happy to merge.

File: docs/tutorials/jax-export.ipynb (hunk 1 above)
Comment (pavithraes) [reply docs/tutorials/jax-export.ipynb:382]: Oh right, I was meaning to clean this but missed it. Will do!

**Comment by pavithraes** [comment]
Thanks, I've addressed your comment. I also included a minor update to keep the structure (installation & troubleshooting) consistent with the proposed updates to the PyTorch export docs.

This is ready for final review & merge. :)

**Review by GleasonK — APPROVED** [review]
